### ISSUE TITLE
Parsing the wrong class name when annotations followed by comments

### ISSUE THREAD

**Issue by functioner** [javaparser/javaparser#3618]
I am using javaparser 3.23.1 and apply it to Apache HBase.
If parsing `RegionObserver.java`:
```
@InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.COPROC)
@InterfaceStability.Evolving
// TODO as method signatures need to break, update to
// ObserverContext<? extends RegionCoprocessorEnvironment>
// so we can use additional environment state that isn't exposed to coprocessors.
public interface RegionObserver {
  // ...
}
```
Then, from the AST, I found that the class name (`ClassOrInterfaceDeclaration.getName()`) of this `RegionObserver` is not "RegionObserver". Instead, it also includes one line of comment before it: "// so we can use additional environment state that isn't exposed to coprocessors."

I'm wondering if the bug is fixed now or how should we fix it?

**Comment by jlerbsc** [comment]
Hi @functioner,
If you want to get only the name of the node you have to use the method getNameAsString because this method returns the identifier of the node (in this case "RegionObserver ").

**Comment by functioner** [comment]
@jlerbsc 
I see. Thanks for the solution.
BTW, I found that `getName()` returns a `SimpleName` object, and the class name is the `identifier` of `SimpleName`. So, what is the meaning of `toString()` in `SimpleName`? I found the implementation of `toString()` is using a `printer` to print the `SimpleName` and I thought it would be the complete code of the whole class or something like that. In our "RegionObserver" case, why does it return a line of comment and the identifier?
